Backend/app/rider-platform/rider-app/Main/src/API/UI/Search.hs

```diff
@@ -122,12 +123,84 @@ fareProductConstructorModifier = \case
 
 search :: Id Person.Person -> SearchReq -> Maybe Version -> Maybe Version -> Maybe Text -> FlowHandler SearchRes
 search personId req mbBundleVersion mbClientVersion device = withFlowHandlerAPI . withPersonIdLogTag personId $ do
+  person <- Person.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
   checkSearchRateLimit personId
+  logDebug $ ("answer of search" :: Text) <> show person.isSimulated
   updateVersions personId mbBundleVersion mbClientVersion
-  (searchId, searchExpiry, routeInfo) <- case req of
-    OneWaySearch oneWay -> oneWaySearch personId mbBundleVersion mbClientVersion device oneWay
-    RentalSearch rental -> rentalSearch personId mbBundleVersion mbClientVersion device rental
-  return $ SearchRes searchId searchExpiry routeInfo
+  if person.isSimulated
+    then do
+      (searchId, searchExpiry, routeInfo) <- case req of
+        OneWaySearch oneWay -> simulatedOneWaySearch personId mbBundleVersion mbClientVersion device oneWay
+        RentalSearch rental -> simulatedRentalSearch personId mbBundleVersion mbClientVersion device rental
+      return $ SearchRes searchId searchExpiry routeInfo
+    else do
+      (searchId, searchExpiry, routeInfo) <- case req of
+        OneWaySearch oneWay -> oneWaySearch personId mbBundleVersion mbClientVersion device oneWay
+        RentalSearch rental -> rentalSearch personId mbBundleVersion mbClientVersion device rental
+      return $ SearchRes searchId searchExpiry routeInfo
```

Comment:
Could you factor this (and all others) out from the main code into something like `SharedLogic.SimulatedFlow.Search/Confirm/etc.`? and make it work like 
```suggestion
SearchSimulate.interceptWhenSimulated person.isSimulated $ do
      (searchId, searchExpiry, routeInfo) <- case req of
        OneWaySearch oneWay -> oneWaySearch personId mbBundleVersion mbClientVersion device oneWay
        RentalSearch rental -> rentalSearch personId mbBundleVersion mbClientVersion device rental
      return $ SearchRes searchId searchExpiry routeInfo
```
I don't think that we need to see this simulated code much